tests/lang_callable/macros/tsymbol_must_match.nim

```diff
@@ -1,68 +1,120 @@
 discard """
-  description: "Generated symbols not matching the definition must be rejected"
+  description: "Symbols not matching the definition must be rejected"
   action: reject
   cmd: "nim check --msgFormat:sexp --filenames:canonical $options $file"
   nimoutFormat: sexp
 """
 
 import std/macros
 
-macro replaceName(kind: static[NimSymKind], def: untyped): untyped =
-  ## Replaces the identifier in the name slot of `def` with a generated symbol
-  ## of the given `kind`.
-  let def = if def.kind == nnkStmtList: def[0] else: def
-
-  def.expectKind(RoutineNodes)
+proc nameExpr(n: NimNode): NimNode =
+  # Extracts the expression containing the replacement name
+  case n.kind
+  of nnkStmtList, nnkStmtListExpr, nnkForStmt:
+    nameExpr n[^1]
+  of nnkConstSection, nnkConstDef:
+    nameExpr n[0]
+  of nnkDiscardStmt:
+    nameExpr n[0]
+  of RoutineNodes:
+    n.name
+  of AtomicNodes:
+    n
```

Comment:
```suggestion
  of nnkStmtList:
    nameExpr n[^1]
  of nkLetSection, nkVarSection, nkConstSection:
    n[0][0]
  of RoutineNodes:
    n.name
```
I would suggest just always using routines and let/var statements as the symbol suppliers. The only reason I sometimes used `nskForVar` was because it cost effectively nothing.